dcs/core/data_repository.py

```diff
@@ -212,3 +215,44 @@ async def serve_envelope(self, *, drs_id: str, public_key: str) -> str:
             raise self.EnvelopeNotFoundError(object_id=drs_id) from error
 
         return envelope
+
+    async def delete_file(self, *, file_id: str) -> None:
+        """Deletes a file from the outbox storage, the internal database and the
+        corresponding secret from the secrets store.
+        If no file or secret with that id exists, do nothing.
+
+        Args:
+            file_id: id for the file to delete.
+        """
+
+        # get secret id
+        drs_object = await self._drs_object_dao.get_by_id(id_=file_id)
+
+        # Call EKSS to remove file secret from vault
+        try:
+            delete_secret_from_ekss(
+                secret_id=drs_object.decryption_secret_id,
+                api_base=self._config.ekss_base_url,
+            )
+        except exceptions.SecretNotFoundError:
```

Comment:
RequestFailedError should also be handled, else the DCS crashes if we have connection issues to the ekss. Same goes for BadResponse code when, e.g. ekss should unexpectedly return 500

Reply:
This is by default, since we don't want to send out a failure event, but also don't want to acknowledge an event where we were not able to delete all information.